apollo-router/src/plugins/traffic_shaping/mod.rs

```diff
@@ -311,6 +303,32 @@ impl TrafficShaping {
         <S as Service<supergraph::Request>>::Future: std::marker::Send,
     {
         ServiceBuilder::new()
+            .map_future_with_request_data(
```

Comment:
I don't think we need to pass the context here as it's already part of the response

Reply:
@bnjjj It's not.
If the timeout or rate-limiting code is returning an error; I don't have access to the response at all since it inside `Ok`.
Also, timeout, in particular, triggers before the response is created, so we need to have some code that passes context to the timeout error somewhere.